### DISCUSSION TITLE
Docstring

### DISCUSSION THREAD

**Discussion by modolol** [MeteoSwiss/dvas#143] [Ideas]
@fpavogt What do you think about such approch to avoid docstings repeat?
I thought that it would be insteresting to modify a little bite the decoraator in order to be able to insert piece of docstring into another one.
[https://softwareengineering.stackexchange.com/questions/386755/sharing-docstrings-between-similar-functions](https://softwareengineering.stackexchange.com/questions/386755/sharing-docstrings-between-similar-functions)

**Comment by fpavogt** [comment]
Yes, that sounds awesome. Were you thinking about a specific set of dvas functions ? The `Profile` strategies come to mind ... at the moment, we have duplicated sets of doctsrings for the `execute()` functions at the core of the strategies, and the corresponding `Profile` class method. Do you think this could help us improve this ? 

It's not clear to me how the "strategy class" docstring would be handled though ...

**Reply by modolol** [reply]
Yes, I was also thinking about the `execute()` methods of the stategies. But also interpreters. We should do some tests to see what it gives. Should we try this for the v0.4 version?

**Reply by fpavogt** [reply]
We could definitely work on this in 0.4, it absolutely counts as "code cleanup". But it's not a necessity either, it's really up to us. These days I'm working on getting the `dvas_recipes` up to speed so we can start to process the multi gdp flights more easily. Which means I'll be staying away from the core dvas routines for a bit: if you wanted to toy with them, we should be able to avoid conflicts. Else, I can also look into it myself once I am done with the recipes ... whatever works best for you !
